src/software/ai/navigator/path_planner/global_path_planner_factory.cpp

```diff
@@ -2,58 +2,120 @@
+                bool parsed =
+                    TbotsProto::MotionConstraint_Parse(enum_value->name(), &constraint);
+                CHECK(parsed) << "Couldn't parse MotionConstraint with value: "
+                              << enum_value->name() << std::endl;
+
+                if (enum_value->options().HasExtension(TbotsProto::dynamic) &&
+                    enum_value->options().GetExtension(TbotsProto::dynamic))
+                {
+                    dynamic_motion_constraints.insert(constraint);
+                    has_dynamic_constraints = true;
+                    break;
+                }
+                static_motion_constraints.emplace(constraint);
             }
         }
 
-        auto obstacles = obstacle_factory.createFromMotionConstraints(
-            motion_constraint_obstacles, world);
-        planners.emplace(std::make_pair(
-            motion_constraint_obstacles,
-            std::make_shared<EnlsvgPathPlanner>(world.field().fieldBoundary(), obstacles,
-                                                ROBOT_MAX_RADIUS_METERS)));
+        if (!has_dynamic_constraints)
+        {
+            auto obstacles = obstacle_factory.createStaticObstaclesFromMotionConstraints(
+                static_motion_constraints, field);
+            planners.emplace(std::make_pair(
+                static_motion_constraints,
+                std::make_shared<EnlsvgPathPlanner>(field.fieldBoundary(), obstacles,
+                                                    ROBOT_MAX_RADIUS_METERS)));
+
+            google::protobuf::RepeatedPtrField<TbotsProto::Obstacles>
+                repeated_obstacle_proto;
+            for (const auto &obstacle : obstacles)
+            {
+                *(repeated_obstacle_proto.Add()) = obstacle->createObstacleProto();
+            }
+
+            motion_constraint_to_obstacles.emplace(
+                std::make_pair(static_motion_constraints, repeated_obstacle_proto));
+        }
     }
 }
 
 std::shared_ptr<const EnlsvgPathPlanner> GlobalPathPlannerFactory::getPathPlanner(
-    const std::set<MotionConstraint> &motion_constraints) const
+    std::set<TbotsProto::MotionConstraint> constraints) const
 {
+    for (const auto &dynamic_constraint : dynamic_motion_constraints)
+    {
+        constraints.erase(dynamic_constraint);
+    }
     try
     {
-        return planners.at(motion_constraints);
+        return planners.at(constraints);
     }
     catch (std::out_of_range &e)
     {
         LOG(WARNING)
             << "GlobalPathPlannerFactory is unable to obtain a path planner for the following motion constraints: ";
-        for (auto constraint : motion_constraints)
+        for (auto constraint : constraints)
         {
             LOG(WARNING) << toString(constraint);
         }
         LOG(WARNING) << "Returning obstacle-free planner.";
     }
     return planners.at({});
 }
+
+google::protobuf::RepeatedPtrField<TbotsProto::Obstacles>
+GlobalPathPlannerFactory::getObstacles(
```

Comment:
```suggestion
GlobalPathPlannerFactory::getStaticObstacles(
```

To be clearer that we ignore the dynamic motion constraints

Reply:
fixed